sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/actuate/autoconfigure/cosmos/CosmosHealthConfiguration.java

```diff
@@ -16,13 +18,13 @@
  */
 @Configuration
 @ConditionalOnClass({ CosmosAsyncClient.class, HealthIndicator.class})
-@AutoConfigureAfter(CosmosAutoConfiguration.class)
+@AutoConfigureAfter(CosmosDataAutoConfiguration.class)
 public class CosmosHealthConfiguration {
 
     @Bean
     @ConditionalOnEnabledHealthIndicator("azure-cosmos")
     public HealthIndicator cosmosHealthContributor(CosmosAsyncClient cosmosAsyncClient) {
-        return new CosmosHealthIndicator(cosmosAsyncClient);
+        return new CosmosHealthIndicator(cosmosAsyncClient, null, null);
```

Comment:
really wired code, why passing into two nulls?

Reply:
and I think most of our health indicators should be `org.springframework.boot.actuate.health.AbstractReactiveHealthIndicator` to take advantage of our sdk's reactive nature.

just like `org.springframework.boot.actuate.cassandra.CassandraDriverReactiveHealthIndicator`
would you please open another issue to have this improvement?

Reply:
Haven't figured this out, the health indicator requires two more parameters, which is not required by the auto-configuration of a cosmos cient. So I'm thinking to expose this two parameters in health indicator's configuration.